r/R/arrow-tabular.R

```diff
@@ -211,6 +211,29 @@ head.ArrowTabular <- head.ArrowDatum
 #' @export
 tail.ArrowTabular <- tail.ArrowDatum
 
+#' @export
+na.fail.ArrowTabular <- function(object, ...){
+  na_count <- sum(purrr::map_int(object$columns, ~.x$null_count))
+  if(na_count > 0){
+    stop("missing values in object")
+  }
+  object
+}
+
+#' @export
+na.omit.ArrowTabular <- function(object, ...){
+  filter_text = paste0(
+    ".data$Filter(",
+    paste0("!is.na(.data$", names(object), ")", collapse = " & "),
+    ")"
+  )
+  filter = rlang::parse_expr(filter_text)
+  rlang::eval_tidy(filter,  rlang::new_data_mask(rlang::env(.data = object)))
+}
```

Comment:
Is how I've implemented this OK?  It feels bad/wrong in a way I can't quite articulate to construct a call to filter this way, or is it alright as I've done it with a data mask etc?

Reply:
(After separate a conversation on Hangouts with @jonkeane , we agreed that this might not be the best way to do this and so I'm going to have a go at implementing this using `purrr` instead.)

Reply:
This looks much cleaner now!